improver/calibration/dataframe_utilities.py

```diff
@@ -318,20 +359,30 @@ def _prepare_dataframes(
         forecast_df = forecast_df[np.logical_or.reduce(indices)]
 
     # Check the percentiles can be considered to be equally space quantiles.
-    _quantile_check(forecast_df)
+    if representation_type == "percentile":
+        _quantile_check(forecast_df)
 
     # Remove forecast duplicates.
-    forecast_df = forecast_df.drop_duplicates(
-        subset=["diagnostic", "forecast_period", "percentile", "time", "wmo_id"],
-        keep="last",
-    )
+    forecast_cols = [
+        "diagnostic",
+        "forecast_period",
+        representation_type,
+        "time",
+        "wmo_id",
+    ]
+    if "station_id" in forecast_df.columns:
+        forecast_cols.append("station_id")
+    forecast_df = forecast_df.drop_duplicates(subset=forecast_cols, keep="last",)
     # Sort to ensure a consistent ordering after removing duplicates.
-    forecast_df = forecast_df.sort_values(
-        by=["blend_time", "percentile", "wmo_id"], ignore_index=True,
-    )
+    sort_cols = ["blend_time", representation_type, "wmo_id"]
+    if "station_id" in forecast_df:
+        sort_cols.append("station_id")
+    forecast_df = forecast_df.sort_values(by=sort_cols, ignore_index=True,)
 
     # Remove truth duplicates.
     truth_cols = ["diagnostic", "time", "wmo_id"]
+    if "station_id" in truth_df.columns:
```

Comment:
I'm not sure about this. If we fill station_id with a null value, then the output cubes will always have an aux coord for station_id. I'm not sure if this would break anything for MO.

Reply:
OK, sounds like this is best left as-is then.